framework/core/js/src/common/Application.tsx

```diff
@@ -130,6 +130,7 @@ export interface ApplicationData {
   locales: Record<string, string>;
   resources: SavedModelData[];
   session: { userId: number; csrfToken: string };
+  maintenanceMode?: 2;
```

Comment:
I know it's a draft PR, but would it be nicer to have a `MaintenanceMode` enum with descriptive names?

Reply:
Better indeed!